ha_vpn_over_interconnect_10GB_attach/main.tf

```diff
@@ -0,0 +1,421 @@
+# HA VPN over Cloud Interconnect 10GB VLAN attachments (Dedicated Interconnect)
+
+# [START cloudinterconnect_ha_vpn_over_interconnect_10gb_attachments]
+
+# Create all resources in the same region, which you can specify at the provider-level
+ provider "google" {
+  project = "your-project"
+  region = "us-east4"  # Region must support creation of all new attachments on Dataplane v2
+}
+
+# VPC Network
+resource "google_compute_network" "network_havpn_ic" {
+  name = "network-havpn-ic"
+  auto_create_subnetworks = false
+  routing_mode = "GLOBAL"
+}
+
+# Subnet
+resource "google_compute_subnetwork" "subnet_havpn_ic" {
+  name = "subnet-havpn-ic"
+  ip_cidr_range = "192.168.1.0/24"
+  network = google_compute_network.network_havpn_ic.self_link
+}
+
+# Begin Cloud Interconnect tier
+# Create Interconnect Cloud Router, specific to HA VPN over Cloud Interconnect
+
+resource "google_compute_router" "ic_router" {
+  name = "ic-router"
+  network = google_compute_network.network_havpn_ic.self_link
+  encrypted_interconnect_router = true
+  bgp {
+    asn = 65000
+  }
+}
+
+# Optional: Reserve regional internal IP ranges to allocate to the HA VPN gateway
+# interfaces. Reserve an internal range for each VLAN attachment.
+
+resource "google_compute_address" "address_vpn_ia_1" {
+  name          = "address-vpn-ia-1"
+  address_type  = "INTERNAL"
+  purpose       = "IPSEC_INTERCONNECT"
+  address       = "192.168.20.0"
+  prefix_length = 29
+  network       = google_compute_network.network_havpn_ic.self_link
+}
+
+resource "google_compute_address" "address_vpn_ia_2" {
+  name          = "address-vpn-ia-2"
+  address_type  = "INTERNAL"
+  purpose       = "IPSEC_INTERCONNECT"
+  address       = "192.168.21.0"
+  prefix_length = 29
+  network       = google_compute_network.network_havpn_ic.self_link
+}
+
+# Create encrypted VLAN attachments
+
+resource "google_compute_interconnect_attachment" "ia_1" {
+  name = "ia-1"
+  router = google_compute_router.ic_router.self_link
+  interconnect = "https://www.googleapis.com/compute/v1/projects/your-project/global/interconnects/interconnect-zone1"
```

Comment:
Is there anyway we can use the data block for this? For example:

```
data "google_project" "project" {
}
resource "google_project_iam_member" "os-login-admin-users" {
  project  = data.google_project.project.project_id
  role = "roles/compute.osAdminLogin"   
  member   = "serviceAccount:service-${data.google_project.project.number}@compute-system.iam.gserviceaccount.com"
}
```

Reply:
https://registry.terraform.io/providers/hashicorp/google/latest/docs/data-sources/project

Reply:
Done and added comments around this